src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,139 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
-    @Override
-    public void add(T value) {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private Object[] elementData;
 
+    public ArrayList() {
+        this.elementData = new Object[DEFAULT_CAPACITY];
     }
 
     @Override
-    public void add(T value, int index) {
+    public boolean add(T value) {
+        if (size == elementData.length) {
+            elementData = grow();
+        }
+        elementData[size] = value;
+        size++;
+        return true;
+    }
 
+    @Override
+    public void add(T value, int index) {
+        rangeCheckForAdd(index);
+        if (size == elementData.length) {
+            elementData = grow();
+        }
+        System.arraycopy(elementData, index,
+                elementData, index + 1,
+                size - index);
+        elementData[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (size + list.size() > elementData.length) {
+            elementData = grow();
+        }
+        for (int i = 0; i < list.size(); i++) {
+            elementData[size++] = list.get(i);
+        }
+    }
 
+    private void checkIndex(int index) {
+        if ((index >= size || index < 0)) {
+            throw new ArrayListIndexOutOfBoundsException(outOfBoundsMsg(index));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return (T) elementData[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index);
+        elementData[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        @SuppressWarnings("unchecked") T oldValue = (T) elementData[index];
+        if ((size - 1) > index) {
+            System.arraycopy(elementData, index + 1, elementData, index, size - index);
+        }
+        elementData[--size] = null;
+        return oldValue;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        int i = 0;
+        if (element == null) {
+            for (; i < size; i++) {
+                if (elementData[i] == null) {
+                    return remove(i);
+                }
+            }
+        } else {
+            for (; i < size; i++) {
+                if (element.equals(elementData[i])) {
+                    return remove(i);
+                }
+            }
+
+        }
+        throw new NoSuchElementException("Could not be removed because no such item "
+                + element);
```

Comment:
<img width="1077" alt="image" src="https://user-images.githubusercontent.com/81781479/202278883-be668ff2-2651-4f81-b0d8-1e40a726ba09.png">

Reply:
![image](https://user-images.githubusercontent.com/11096426/202292171-f73443ad-f68b-4cba-9943-2916664d0348.png)

Reply:
> <img alt="image" width="1077" src="https://user-images.githubusercontent.com/81781479/202278883-be668ff2-2651-4f81-b0d8-1e40a726ba09.png">

it doesn't work

Reply:
![image](https://user-images.githubusercontent.com/11096426/202306146-e70e1e89-c7f8-41e3-85c4-490c1a6a67e0.png)
maybe you wanted that